[CLANG_X] Undefined references to undefined reference to `perftools::EdmEventSize` members

After https://github.com/cms-sw/cmssw/pull/47248 was merged, CLANG_X IBs are failing to build with the following error [link](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc12/CMSSW_15_1_CLANG_X_2025-02-09-2300/PerfTools/EdmEvent):

```
>> Building binary edmEventSize
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/llvm/19.1.6-bde731e5eaefc2cb5ab74d026aee87b7/bin/clang++ -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-c99-extensions -Wno-c++11-narrowing -D__STRICT_ANSI__ -Wno-unused-private-field -Wno-unknown-pragmas -Wno-unused-command-line-argument -Wno-unknown-warning-option -ftemplate-depth=512 -Wno-error=potentially-evaluated-expression -Wno-tautological-type-limit-compare -Wno-vla-cxx-extension -fsized-deallocation --gcc-toolchain=/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -fPIC    tmp/el8_amd64_gcc12/src/PerfTools/EdmEvent/bin/edmEventSize/edmEventSize.cpp.o -Wl,-E -Wl,--hash-style=gnu -Wl,--as-needed -Wl,-z,noexecstack -L/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/97b0a163379b222df2d5f7ddf2ec37b7/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_CLANG_X_2025-02-09-2300/biglib/el8_amd64_gcc12 -L/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/97b0a163379b222df2d5f7ddf2ec37b7/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_CLANG_X_2025-02-09-2300/lib/el8_amd64_gcc12 -L/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/97b0a163379b222df2d5f7ddf2ec37b7/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_CLANG_X_2025-02-09-2300/external/el8_amd64_gcc12/lib -L/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/97b0a163379b222df2d5f7ddf2ec37b7/opt/cmssw/el8_amd64_gcc12/cms/cmssw/CMSSW_15_1_CLANG_X_2025-02-09-2300/static/el8_amd64_gcc12 -lFWCoreFWLite -lDataFormatsCommon -lFWCoreMessageLogger -lDataFormatsProvenance -lFWCorePluginManager -lFWCoreReflection -lFWCoreUtilities -lPerfToolsEdmEvent -lTreePlayer -lGraf3d -lPostscript -lGpad -lGraf -lPhysics -lHist -lMatrix -lTree -lNet -lThread -lMathCore -lRIO -lboost_program_options -lCore -lboost_thread -lboost_date_time -lpcre -lbz2 -luuid -ltbb -llzma -lz -lfmt -lcms-md5 -lcrypt -ldl -lrt -lstdc++fs -ltinyxml2 -o tmp/el8_amd64_gcc12/src/PerfTools/EdmEvent/bin/edmEventSize/edmEventSize
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/PerfTools/EdmEvent/bin/edmEventSize/edmEventSize.cpp.o: in function `void processRecord<(perftools::EdmEventMode)1>(boost::program_options::variables_map const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)':
  edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x1e): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::EdmEventSize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x167): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::formatNames()'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x297): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::sortAlpha()'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x57b): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::dumpJson(std::ostream&) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x593): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::dump(std::ostream&, bool) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0xa02): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::dumpJson(std::ostream&) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0xa1b): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::dump(std::ostream&, bool) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE1EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x1017): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)1>::produceHistos(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int) const'
 /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/PerfTools/EdmEvent/bin/edmEventSize/edmEventSize.cpp.o: in function `void processRecord<(perftools::EdmEventMode)0>(boost::program_options::variables_map const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)':
  edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x1e): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::EdmEventSize(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x167): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::formatNames()'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x297): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::sortAlpha()'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x57b): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::dumpJson(std::ostream&) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x593): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::dump(std::ostream&, bool) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0xa02): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::dumpJson(std::ostream&) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0xa1b): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::dump(std::ostream&, bool) const'
   /data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: edmEventSize.cpp:(.text._Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b[_Z13processRecordILN9perftools12EdmEventModeE0EEvRKN5boost15program_options13variables_mapERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESE_b]+0x1017): undefined reference to `perftools::EdmEventSize<(perftools::EdmEventMode)0>::produceHistos(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int) const'
 clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reverting that PR allows compilation to pass.

--- a/PerfTools/EdmEvent/interface/EdmEventSize.h
+++ b/PerfTools/EdmEvent/interface/EdmEventSize.h
@@ -95,10 +95,8 @@ namespace perftools {
     Records m_records;
   };
 
-  template class perftools::EdmEventSize<perftools::EdmEventMode::Leaves>;
-
-  template class perftools::EdmEventSize<perftools::EdmEventMode::Branches>;
-
+  extern template class perftools::EdmEventSize<perftools::EdmEventMode::Leaves>;
+  extern template class perftools::EdmEventSize<perftools::EdmEventMode::Branches>;
 }  // namespace perftools
 
 #endif  // PerfTools_EdmEventSize_H

--- a/PerfTools/EdmEvent/src/EdmEventSize.cc
+++ b/PerfTools/EdmEvent/src/EdmEventSize.cc
@@ -423,4 +423,6 @@ namespace perftools {
     }
   }
 
+  template class perftools::EdmEventSize<perftools::EdmEventMode::Leaves>;
+  template class perftools::EdmEventSize<perftools::EdmEventMode::Branches>;
 }  // namespace perftools

